internal/diff/diff.go

```diff
@@ -69,12 +68,11 @@ func writeTempFile(dir, prefix string, data []byte) (string, error) {
 		return "", err
 	}
 	_, err = file.Write(data)
-	if err1 := file.Close(); err == nil {
-		err = err1
-	}
 	if err != nil {
-		_ = os.Remove(file.Name())
-		return "", err
+		return file.Name(), err
```

Comment:
The problem with this centralization is that this seems weird to me - on error, you're still returning me the name of the file, but what does that mean? Does the file exist? I would expect the function that creates the temp file to do the cleanup itself upon error before returning, hence the existing logic.